docs/howto_port_a_command_to_engine_p.md

```diff
@@ -0,0 +1,35 @@
+# How To Port Old Engine Syntax to the Latest and Greatest
+
+## engine-p
+
+Even in the new codebase of nu things change and we have some old code to clean up.
+This guide walks you through how to port old engine syntax to engine-p.
+
+1. change into the commands source directory
+
+```sh
+cd crates/nu-command/src/
+```
+
+2. search for old syntax using ripgrep (`rg`)
+
+```sh
+rg --type rust --files-with-matches 'ActionStream|run_with_actions'
+```
+
+3. update the old syntax engine-p syntax
+  - For a smaller example PR see [#3794](https://github.com/nushell/nushell/pull/3794/files)
+  - For a more involved example PR see [#3649](https://github.com/nushell/nushell/pull/3649/files)
+
+In many cases this is changing the function name and signature.
+The function name goes from `run_with_actions` to `run`.
+The signature goes from `Result<ActionStream, ShellError>`
+to `Result<OutputStream, ShellError>`.
```

Comment:
You might want to call out when you wouldn't be able to updated to `OutputStream`, eg when there are Actions that might also be part of the output.

Reply:
@jntrnr good point. Could you share an example PR or link to a source file of what you were thinking of? I'm having trouble finding or coming up with an example of this.

Reply:
@efx - One example is a command like `open` couldn't be converted because it can output an Action that will convert the data from one form to another.